working-groups/security/threat-model.md

```diff
@@ -0,0 +1,81 @@
+# Knative Threat Model
+
+This document describes the Knative threat model. When vulnerabilities are
+reported to the project, we consult this document to determine whether the
+report describes a potential exploit, and if so to determine the severity of
+the exploit. As we develop new features, we consult this document to consider
+their impact on the threat model.
+
+_Status:_ Work in Progress. This document does not yet represent the full
+threat model envisioned by the knative community, but we have to start
+somewhere.
+
+## Tenancy Model
+
+There are three common tenancy models currently in use in the Kubernetes
+Ecosystem (See [Three Tenancy Models for Kubernetes](https://kubernetes.io/blog/2021/04/15/three-tenancy-models-for-kubernetes/)).
+Summarising here, in the first, a cluster is fully owned by a single tenant
+(Cluster-as-a-service).
+In the second, a cluster is partitioned between multiple tenants with the
+Namespace forming a security boundary between tenants of the cluster
+(Namspace-as-a-service).
+In the third, resources such as nodes and schedulers are shared, but the
+tenants, are presented with a logical cluster and can perform global operations
+such as installing CRDs and managing namespaces (Control-plane-as-a-service).
+
+Currently, Knative assumes a single tenant per cluster (Cluster-as-a-service)
+as its security model.
+
+It is possible, with extra work, to secure Knative for use in
+Namespace-as-a-service environments, for example by setting up appropriate
+NetworkPolicy, using appropriate container isolation techniques, and filtering
+requests to shared components such as the activator. The project aims to
+eventually support secure use in Namespace-as-a-service environments out of the
+box, and is very receptive to PRs that improve support for
+Namespace-as-a-service clusters, but we do not currently view exploits in this
+environment as vulnerabilities for the purpose of our disclosure policy.
+
+## Trusted and untrusted workloads
+
+Knative is often used to implement platforms which permit users to submit code
```

Comment:
Let's distinguish between _users_ who have access to only the http endpoints exposed outside the cluster and _developers_ who have RBAC access to update some objects on the kubernetes cluster.